owlkettle/adw.nim

```diff
@@ -1286,6 +1289,156 @@ renderable AboutWindow {.since: AdwVersion >= (1, 2).}:
+  
+  let event = EventObj[proc()]()
+  let data = allocSharedCell(event)
+  data.callback = clickedHandler
+  data.handler = g_signal_connect(toast, "button-clicked".cstring, clickCallback, data)
+
+proc toGtk(toast: Toast): AdwToast =
+  result = adw_toast_new(toast.title.cstring)
+  if toast.buttonLabel != "":
+    adw_toast_set_button_label(result, toast.buttonLabel.cstring)
+  adw_toast_set_priority(result, toast.priority)
+  adw_toast_set_timeout(result, toast.timeout.cuint)
+  
+  # Set Dismissal Handler
+  if not toast.dismissalHandler.isNil():
+    result.connectToastDismissalHandler(toast.dismissalHandler)
+  
+  when AdwVersion >= (1, 2):
+    if not toast.customTitle.isNil():
+      let customTitleWidget = toast.customTitle.build().unwrapInternalWidget()
+      adw_toast_set_custom_title(result, customTitleWidget)
+
+    # Set Clicked Handler
+    if not toast.clickedHandler.isNil():
+      result.connectToastClickedHandler(toast.clickedHandler)
+  
+  when AdwVersion >= (1, 4):
+    adw_toast_set_use_markup(result, toast.useMarkup.cbool)
+
+type ToastQueue* = ref object
+  toasts: seq[Toast]
+  addToOverlay: proc(toast: Toast) {.closure.}
+
+proc isInitialized(queue: ToastQueue): bool = not queue.addToOverlay.isNil()
+proc newToastQueue*(): ToastQueue = ToastQueue()
+proc add*(queue: ToastQueue, toast: Toast) = 
+  queue.toasts.add(toast)
+    
+proc add*(queue: ToastQueue, toasts: openArray[Toast]) =
+  for toast in toasts:
+    queue.add(toast)
+
+proc clear(queue: ToastQueue) = queue.toasts = @[]
+
+renderable ToastOverlay of BaseWidget:
+  ## An overlay to display Toast messages that can be dismissed manually and automatically!<br>
+  ## Use `newToast` to create a `Toast`.
+  ## `Toast` has the following properties that can be assigned to:
+  ## - actionName
+  ## - actionTarget
+  ## - buttonLabel: If set, the Toast will contain a button with this string as its text. If not set, the Toast will not contain a button.
+  ## - detailedActionName
+  ## - priority: Defines the behaviour of the toast. `ToastPriorityNormal` will put the toast at the end of the queue of toasts to display. `ToastPriorityHigh` will display the toast **immediately**, ignoring any others.
+  ## - timeout: The time in seconds after showing the toast after which it is dismissed automatically. Disables automatic dismissal if set to 0. Defaults to 5. 
+  ## - title: The text to display in the toast. Gets hidden if customTitle is set.
+  ## - customTitle: A Widget to display in the toast. Causes title to be hidden if it is set. Only available when compiling for Adwaita version 1.2 or higher.
+  ## - dismissalHandler: An event-handler proc that gets called when this specific toast gets dismissed
+  ## - clickedHandler: An event-handler proc that gets called when the User clicks on the toast's button that appears if `buttonLabel` is defined. Only available when compiling for Adwaita version 1.4 or higher.
```

Comment:
Should only list the fields that are actually implemented.

Reply:
Removed actionName, actionTarget and detailedActionName as discussed above.